pkg/util/hostname.go

```diff
@@ -249,12 +276,22 @@ func GetHostname() (string, error) {
 		err = nil
 	}
 
-	cache.Cache.Set(cacheHostnameKey, hostName, cache.NoExpiration)
-	setHostnameProvider(provider)
+	hostnameData := saveHostnameData(cacheHostnameKey, hostName, provider)
 	if err != nil {
 		expErr := new(expvar.String)
 		expErr.Set(fmt.Sprintf(err.Error()))
 		hostnameErrors.Set("all", expErr)
 	}
-	return hostName, err
+	return hostnameData, err
+}
+
+// isHostnameCanonicalForIntake returns true if the intake will use the hostname as canonical hostname.
+func isHostnameCanonicalForIntake(hostname string) bool {
+	// Intake uses instance id for ec2 default hostname except for Windows.
+	if ec2.IsDefaultHostnameForIntake(hostname) {
+		if _, err := ec2.GetInstanceID(); err == nil {
+			return config.Datadog.GetBool("hostname_force_config_as_canonical")
```

Comment:
nit: `config.Datadog.GetBool("hostname_force_config_as_canonical")` should be outside of this function. The way I understand it, this function returns whether the `hostname` field of the metadata payload will be used as canonical hostname by the intake, and that doesn't depend on the value of `hostname_force_config_as_canonical`

We should only log the warning when `hostname_force_config_as_canonical` is set to false though, agreed, but let's have this check outside of this function

Reply:
Fixed.